### PR TITLE
decrypt seen users only

### PR DESCRIPTION
**Pull request by butonic**
It does not make sense to iterate aver all backends individually, now that we have the account table. Saves a lot of time when having an ldap with 100k users. Also romoves the dependency on a live ldap server.

### CODE REVIEW

**Review by sharidas — APPROVED** [review]
I have verified this change and it worked like charm. This change would definitely improve the performance. And yes it doesn't make sense to loop through users who didn't logged in once. A double :+1:  from my side. If the tests are green, then we should take this change.

**Comment by PVince81** [comment]
@butonic why "To develop", anything missing ?

**Comment by butonic** [comment]
@PVince81 This PR was the reason for me asking about how to test callback stuff. The last amend allowed passing the tests ... will check the test coverage tomorrow.

**Comment by PVince81** [comment]
@butonic any update on the tests?

**Comment by PVince81** [comment]
@butonic do you intend to finish this or should someone else take over ?

**Comment by PVince81** [comment]
@sharidas mind taking over and helping to finish this ?

File: lib/private/Encryption/DecryptAll.php [hunk 1]
```diff
@@ -164,6 +164,9 @@ protected function decryptAllUsersFiles($user = '') {
 		if ($user === '') {
 			$userNo = 1;
 			$numberOfUsers = $this->userManager->countSeenUsers();
+			if ($numberOfUsers === 0) {
```
Comment (sharidas) [inline lib/private/Encryption/DecryptAll.php]: In a scenario where none of the users are seen ( i.e, none of the users are logged in ), then `numberOfUsers` become zero. And guess we don't have to call the `callForSeenUsers`. I have tested in a scratch installation with 2 users who were not logged in and the decrypt-all command is called.

**Review by PVince81 — APPROVED** [review]
:+1: please backport

**Comment by sharidas** [comment]
Backport PR: https://github.com/owncloud/core/pull/30640

**Pull request merged**